syntax = "proto3";

package files;

service FileService {
  rpc UploadFile(UploadFileRequest) returns (FileResponse);
  rpc DownloadFile(DownloadFileRequest) returns (stream FileChunk);
  rpc RemoveFile(DownloadFileRequest) returns (FileResponse);
}

message UploadFileRequest {
  string file_name = 1;
  bytes file_content = 2;
}

message DownloadFileRequest {
  string file_name = 1;
}

message FileResponse {
  string file_name = 1;
  string message = 2;
}

message FileChunk {
  bytes chunk_data = 1;
}
